app/assets/javascripts/lit/sources.js.coffee

```diff
@@ -0,0 +1,26 @@
+$ ->
+  sourceId = $('#source_id').attr('value')
+  apiKey = $('#api_key').attr('value')
+
+  updateFunc = ->
+    $.ajax "/lit/api/v1/sources/" + sourceId + "/sync_complete",
+      type: 'GET'
+      headers:
+        'Authorization': "Token token=\"#{apiKey}\""
+      success: (xml, textStatus, xhr) ->
+        if xhr.responseJSON.sync_complete
+          $('.loading').addClass('loaded').removeClass('loading')
+          clearInterval(interval)
+          location.reload()
+      statusCode:
+        404: ->
+          $('.loading').text('Could not update synchronization status, please try refreshing page')
+        401: ->
+          $('.loading').text('Invalid API key, please check source settings')
```

Comment:
@mlitwiniuk I don't know if I'm not missing something here, but I think that in `SourcesController` the only way to get a 401 is by not providing a valid API key.

@Bonias I can see the point, will investigate the issue. Indeed in this pull request we're dealing with two different APIs which leads to confusion at times.